### PR TITLE
Use pydantic-core==0.29.1

### PR DESCRIPTION
**Pull request by dmontagu**
I also want to add custom init before we do the next alpha release of pydantic but I'll do that in a separate PR (soon to follow) to make it easier to review its tests/etc.

Selected Reviewer: @adriangb

### CODE REVIEW

**Comment by dmontagu** [comment]
please review

File: pdm.lock [hunk 1]
```diff
@@ -1144,88 +1144,88 @@ content_hash = "sha256:b0f4d25f97b612604c8f853c414dc17ac281a6b4bbd14972bdb8e945a
...
-    {url = "https://files.pythonhosted.org/packages/ff/01/c632172ae41b889bcdafe9e7258507d76bb0fa86cfd92011eece069df2ca/pydantic_core-0.28.1-cp311-cp311-macosx_11_0_arm64.whl", hash = "sha256:b469df56352af3c8409977fed98e6faad2b81b157fd1fe7daec89a6aa7b9e172"},
-    {url = "https://files.pythonhosted.org/packages/ff/14/da5d6cecf2cb6ad5af36d5a7593b87d6415ed1412a9f49116e8c2a760ae4/pydantic_core-0.28.1-cp310-cp310-manylinux_2_5_i686.manylinux1_i686.whl", hash = "sha256:80433de8f4334c3acbfaacf6de8cd2c6f57d71068a7418b2be4e08465b6c1e37"},
-    {url = "https://files.pythonhosted.org/packages/ff/7d/2310ebe71aaedbefbe00886107a140fa9103e10f8fa6d5ebdb413bdd75d4/pydantic_core-0.28.1-cp311-cp311-manylinux_2_17_aarch64.manylinux2014_aarch64.whl", hash = "sha256:82b54ec2ab627ddb78aa9b8c23d0a525308c95b68d4316145dc925751424c346"},
+"pydantic-core 0.29.0" = [
+    {url = "https://files.pythonhosted.org/packages/02/49/823072252731176db03873ab4679b2e80d7cceb5c9e3e0b6ec723f31c8bd/pydantic_core-0.29.0-cp311-cp311-musllinux_1_1_x86_64.whl", hash = "sha256:5b4ca92cac896c8e13b12e42109304ae385e48dde4b321d04238ba56997063c0"},
```
Comment (samuelcolvin) [inline pdm.lock:1148]: looks like you refreshed the whole file, better in general to do `pdm add pydantic-core==0.29.0` and it won't update everything.

**Review by samuelcolvin — APPROVED** [review]
otherwise LGTM.

File: pdm.lock (hunk 1 above)
Comment (dmontagu) [reply pdm.lock:1148]: that's what I did, I think there's just a lot of pydantic core lines

Comment (adriangb) [reply pdm.lock:1148]: One line per wheel

**Comment by dmontagu** [comment]
Closing in favor of #5703

File: pdm.lock (hunk 1 above)
Comment (samuelcolvin) [reply pdm.lock:1148]: 🤦 🙏

**Pull request closed without merging**